### PR TITLE
123 Poll request body json protocol

### LINKED ISSUE #123: PollRequestBody - JSONRequestBody protocol
ℹ️ This issue is part of Epic #62 

## Objective

Implement JSONRequestBody to convert the PollRequestBody to a valid JSON to send via API

## Checklist

* parse: input a PollRequestBody and gives a "VALID_JSON"

### PR DESCRIPTION
**Pull request by SergioGueruiz**
Fixes #123 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: lib/types/poll_request_body.ex [hunk 1]
```diff
@@ -23,4 +25,30 @@ defmodule Kadena.Types.PollRequestBody do
 
   def new(%Base64UrlsList{} = request_keys), do: %__MODULE__{request_keys: request_keys}
   def new(_request_keys), do: {:error, [request_keys: :not_a_list]}
+
+  defimpl JSONRequestBody do
+    alias Kadena.Types.{
+      Base64Url,
+      Base64UrlsList,
```
Comment (boterop) [inline lib/types/poll_request_body.ex]: it is imported in the top of the struct, you should remove this one

File: lib/types/poll_request_body.ex [hunk 2]
```diff
@@ -23,4 +25,30 @@ defmodule Kadena.Types.PollRequestBody do
...
+      PollRequestBody
+    }
+
+    @type key_list :: {:ok, list()}
+    @type base64_urls :: Base64UrlsList.t()
+
+    @impl true
+    def parse(%PollRequestBody{request_keys: key_list}) do
+      with {:ok, key_list} <- extract_keys(key_list) do
```
Comment (boterop) [inline lib/types/poll_request_body.ex]: IMO this `with case` is unnecessary

File: lib/types/poll_request_body.ex [hunk 3]
```diff
@@ -23,4 +25,30 @@ defmodule Kadena.Types.PollRequestBody do
...
+        %{request_keys: key_list}
+        |> MapCase.to_camel!()
+        |> Jason.encode!()
+      end
+    end
+
+    @spec extract_keys(base64_urls()) :: key_list()
+    defp extract_keys(%Base64UrlsList{urls: list}) do
```
Comment (boterop) [inline lib/types/poll_request_body.ex]: you could return in the `extract_keys` function the list without the :ok

**Review by boterop — CHANGES_REQUESTED** [review]

File: lib/types/poll_request_body.ex [hunk 4]
```diff
@@ -23,4 +25,26 @@ defmodule Kadena.Types.PollRequestBody do
...
+      Base64Url,
+      PollRequestBody
+    }
+
+    @type urls :: list(Base64Url.t())
+    @type base64_urls :: Base64UrlsList.t()
+
+    @impl true
+    def parse(%PollRequestBody{request_keys: key_list}) do
+      %{request_keys: extract_keys(key_list)}
```
Comment (boterop) [inline lib/types/poll_request_body.ex]: IMO could be better if the extract_keys is assigned to a variable, or in other pipe

File: lib/types/poll_request_body.ex [hunk 5]
```diff
@@ -23,4 +25,26 @@ defmodule Kadena.Types.PollRequestBody do
...
+      |> MapCase.to_camel!()
+      |> Jason.encode!()
+    end
+
+    @spec extract_keys(base64_urls()) :: urls()
+    defp extract_keys(%Base64UrlsList{urls: list}) do
```
Comment (boterop) [inline lib/types/poll_request_body.ex]: this code is just one line. please use `, do: `

**Review by boterop — CHANGES_REQUESTED** [review]

**Review by boterop — APPROVED** [review]

**Review by einerzg — APPROVED** [review]

**Pull request merged**